solara/components/togglebuttons.py

```diff
@@ -19,13 +20,74 @@ def _get_button_value(button: reacton.core.Element):
     return value
 
 
+@overload
 @solara.value_component(None)
 def ToggleButtonsSingle(
-    value: Union[None, T, solara.Reactive[T]] = None,
+    value: Union[T, solara.Reactive[T]],
+    values: List[T] = ...,
+    children: List[reacton.core.Element] = ...,
+    on_value: Optional[Callable[[T], None]] = ...,
+    dense: bool = ...,
+    mandatory: Literal[True] = ...,
+    classes: List[str] = ...,
+    style: Union[str, Dict[str, str], None] = ...,
+) -> reacton.core.ValueElement[v.BtnToggle, T]:
+    ...
+
+
+@overload
+@solara.value_component(None)
+def ToggleButtonsSingle(
+    value: Union[Optional[T], solara.Reactive[Optional[T]]] = None,
+    values: List[T] = ...,
+    children: List[reacton.core.Element] = ...,
+    on_value: Optional[Callable[[T], None]] = ...,
+    dense: bool = ...,
+    mandatory: Literal[False] = ...,
+    classes: List[str] = ...,
+    style: Union[str, Dict[str, str], None] = ...,
+) -> reacton.core.ValueElement[v.BtnToggle, T]:
+    ...
+
+
+@overload
+@solara.value_component(None)
+def ToggleButtonsSingle(
+    value: Union[T, solara.Reactive[T]],
+    values: List[T] = ...,
+    children: List[reacton.core.Element] = ...,
+    on_value: Optional[Callable[[T], None]] = ...,
+    dense: bool = ...,
+    mandatory: Literal[True] = ...,
+    classes: List[str] = ...,
+    style: Union[str, Dict[str, str], None] = ...,
+) -> reacton.core.ValueElement[v.BtnToggle, T]:
+    ...
+
+
+@overload
+@solara.value_component(None)
+def ToggleButtonsSingle(
+    value: Union[Optional[T], solara.Reactive[Optional[T]]] = None,
+    values: List[T] = ...,
+    children: List[reacton.core.Element] = ...,
+    on_value: Optional[Callable[[T], None]] = ...,
+    dense: bool = ...,
+    mandatory: Literal[False] = ...,
+    classes: List[str] = ...,
+    style: Union[str, Dict[str, str], None] = ...,
+) -> reacton.core.ValueElement[v.BtnToggle, T]:
+    ...
+
+
```

Comment:
```suggestion
@overload
@solara.value_component(None)
def ToggleButtonsSingle(
    value: Union[T, solara.Reactive[T]],
    values: List[T] = ...,
    children: List[reacton.core.Element] = ...,
    on_value: Optional[Callable[[T], None]] = ...,
    dense: bool = ...,
    mandatory: Literal[True] = ...,
    classes: List[str] = ...,
    style: Union[str, Dict[str, str], None] = ...,
) -> reacton.core.ValueElement[v.BtnToggle, T]:
    ...


@overload
@solara.value_component(None)
def ToggleButtonsSingle(
    value: Union[Optional[T], solara.Reactive[Optional[T]]] = None,
    values: List[T] = ...,
    children: List[reacton.core.Element] = ...,
    on_value: Optional[Callable[[T], None]] = ...,
    dense: bool = ...,
    mandatory: Literal[False] = ...,
    classes: List[str] = ...,
    style: Union[str, Dict[str, str], None] = ...,
) -> reacton.core.ValueElement[v.BtnToggle, T]:
    ...


```

We only need two overloads, I think, for mandatory is True and False.